Settings

Shiran1212 opened the pull request:
This pull request includes several important changes to the `app` module, focusing on dependency updates, permissions, and significant additions to the Cloudinary API integration and local database management.


https://github.com/user-attachments/assets/cf8c5ad5-9143-4d97-bdc0-ae3464d29aae



### Dependency Updates:
* Added `circleimageview` and `picasso` libraries to `app/build.gradle.kts` for image handling. [[1]](diffhunk://#diff-8cff73265af19c059547b76aca8882cbaa3209291406f52df1dafbbc78e80c46R67) [[2]](diffhunk://#diff-8cff73265af19c059547b76aca8882cbaa3209291406f52df1dafbbc78e80c46L79-R80)

### Permissions and Manifest Updates:
* Updated `AndroidManifest.xml` to include new permissions for network state and internet access, and added application icons and window soft input mode.

### Cloudinary API Integration:
* Added `CloudinaryApi` interface for image upload functionality.
* Implemented `CloudinaryClient` object to configure and create the Retrofit client for Cloudinary API.
* Created `CloudinaryService` class to handle image upload logic, including file handling and API calls.
* Defined `CloudinaryResponse` data class to model the response from Cloudinary API.

### Local Database Enhancements:
* Updated `AppLocalDatabase` to include `CommentEntity` and added a fallback to destructive migration. [[1]](diffhunk://#diff-7f8df9d16798412d4f0d8f6a427a892566b7b8f5a4305a06319738d2c0f15edaL8-R17) [[2]](diffhunk://#diff-7f8df9d16798412d4f0d8f6a427a892566b7b8f5a4305a06319738d2c0f15edaL27-R31)
* Added `CommentDao` interface to manage comment-related database operations.

### Code Refactoring:
* Refactored imports to use `common` package for shared models in various files. [[1]](diffhunk://#diff-7f82c24fbe5a8d365e0e584c9a627658e05aea387233f3954af76e84d831269eL4-R4) [[2]](diffhunk://#diff-9fa1ab3f9547a9daecf0ccd89d8b369b8a0b42749317c490a2fdfd443f17ca3bL6-R6) [[3]](diffhunk://#diff-826155a5f25192a4d95496339dbc310f5a723c2f7e5774bd0d59d9c3863ac770L3-R5) [[4]](diffhunk://#diff-2aab4604ea2f5faadd4fc777ec7b54026bbaf23995fdf21efb874ba19b4a42c0L4-R4)
* Removed `EventRepository` class, likely due to a shift in data handling strategy.

These changes enhance the app's capabilities in handling images, managing permissions, and improving database operations, while also cleaning up and organizing the codebase.

Shiran1212 committed:
login view

Shiran1212 committed:
login view

Shiran1212 committed:
settings

AsafL910 commented on app/src/main/res/drawable/circle_background_with_shadow.xml:
```diff
@@ -0,0 +1,17 @@
+
+    <item>
+        <shape android:shape="oval">
+            <solid android:color="#C9C9C9" />
```
can you use theme colors here? and everywhere else?

AsafL910 commented on app/src/main/res/layout/fragment_settings.xml:
```diff
@@ -1,15 +1,238 @@
 <?xml version="1.0" encoding="utf-8"?>
```
add screenshots so we can see what it looks like

Shiran1212 committed:
settings

LiorVainer committed:
refactor: reorganize models and data classes for better structure and clarity

LiorVainer committed:
Merged new Event and Event Entity Location Types

LiorVainer committed:
Merge branch 'main' of https://github.com/RonyL02/event-radar into feature/comments-integration

LiorVainer committed:
Add Cloudinary image upload functionality and update event model

- Implement Cloudinary API client and response data class for image uploads.
- Add new drawable resources for UI elements.
- Update EventEntity to use type code instead of name.
- Modify LoginViewModel to use singleton UserRepository instance.
- Adjust layout dimensions in fragment_event_card_shimmer.xml.
- Enhance Comment model with image support and Firestore conversion.
- Update AndroidManifest to include necessary permissions.
- Refactor ViewModelFactory to accommodate CommentsRepository.

LiorVainer committed:
- Made comments fetching for events much faster 43s to 2.2s

- Refactor data classes and remove unnecessary comments for clarity

LiorVainer committed:
Update app icons and layout for comments section

LiorVainer committed:
Added Users Comments List,
- Refactor layout padding
- add new icons
- enhance comment data model

LiorVainer committed:
Refactor padding in event card layouts and remove unnecessary margin from user comment row

LiorVainer committed:
Enhance UI and functionality: update button styles, add off-white color, sync comments, and improve layout for comments section

LiorVainer committed:
Improve comment display logic to handle blank content gracefully

LiorVainer committed:
Merge branch 'feature/comments-integration' into settings

# Conflicts:
#	app/build.gradle.kts
#	app/src/main/AndroidManifest.xml
#	app/src/main/res/drawable/ic_edit.xml
#	app/src/main/res/mipmap-anydpi-v26/ic_launcher.xml
#	app/src/main/res/mipmap-anydpi-v26/ic_launcher_round.xml
#	app/src/main/res/mipmap-hdpi/ic_launcher.webp
#	app/src/main/res/mipmap-hdpi/ic_launcher_round.webp
#	app/src/main/res/mipmap-mdpi/ic_launcher.webp
#	app/src/main/res/mipmap-mdpi/ic_launcher_round.webp
#	app/src/main/res/mipmap-xhdpi/ic_launcher.webp
#	app/src/main/res/mipmap-xhdpi/ic_launcher_round.webp
#	app/src/main/res/mipmap-xxhdpi/ic_launcher.webp
#	app/src/main/res/mipmap-xxhdpi/ic_launcher_round.webp
#	app/src/main/res/mipmap-xxxhdpi/ic_launcher.webp
#	app/src/main/res/mipmap-xxxhdpi/ic_launcher_round.webp
#	gradle/libs.versions.toml

LiorVainer committed:
Refactor User model and initialize LiveData in EventViewModel to improve data handling

LiorVainer committed:
- Added edit profile option with connection to firebase and cloudinary

Refactor UserRepository to accept context and update UserViewModel to use the new repository initialization

LiorVainer committed:
Add user status check and logout functionality in settings
